crates/re_data_source/src/data_loader/loader_external.rs

```diff
@@ -239,14 +277,23 @@ impl crate::DataLoader for ExternalLoader {
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
 fn decode_and_stream<R: std::io::Read>(
     filepath: &std::path::Path,
     tx: &std::sync::mpsc::Sender<crate::LoadedData>,
+    tx_is_sending_data: std::sync::mpsc::SyncSender<()>,
     decoder: re_log_encoding::decoder::Decoder<R>,
 ) {
     re_tracing::profile_function!(filepath.display().to_string());
 
+    let mut is_waiting_for_data = true;
+
     for msg in decoder {
+        if is_waiting_for_data {
+            is_waiting_for_data = false;
+            tx_is_sending_data.send(()).ok();
+        }
```

Comment:
Wouldn't it be a lot simple to just use an `Arc<AtomicBool>` for this communication?